src/multi.c

```diff
@@ -83,7 +83,7 @@ void queueMultiCommand(client *c) {
 void discardTransaction(client *c) {
     freeClientMultiState(c);
     initClientMultiState(c);
-    c->flags &= ~(CLIENT_MULTI|CLIENT_DIRTY_CAS|CLIENT_DIRTY_EXEC);
+    c->flags &= ~(CLIENT_DENY_BLOCKING|CLIENT_MULTI|CLIENT_DIRTY_CAS|CLIENT_DIRTY_EXEC);
```

Comment:
I think MULTI should not be allowed inside LUA or RM_Call, I can check (inside multi) that the CLIENT_DENY_BLOCKING flag is on and if it does return an error (as you suggested). Or I can introduce a new flag that suggests that the connection is an internal connection (i.e there is no real client behind it and its executed from a script or from RM_Call), if this flag is on we can not allows executing commands like MULTI/SUBSCRIBE/MONITOR. What do you think?

Reply:
summary of a chat we had:
let's block MULTI/SUBSCRIBE/MONITOR when the DENY_BLOCING is set. with the exception of allowing SUBSCRIBE in LUA (despite the DENY_BLOCKING flag).
and comment why..

also, we want to set the DENY_BLOCKING flag in AOF client (can't be blocked), but allow it (and add a comment) in the master client (not because of commands like BLPOP, but rather because of modules that decide to process a command in the background).